src/html/scan.js

```diff
@@ -514,7 +514,28 @@ function fileDragHover(e) {
 function fileSelectHandler(e) {
   fileDragHover(e);
   const files = e.target.files || e.dataTransfer.files;
-  uploadFile(files[0]);
+  const fileExt = files[0].name.split('.').pop();
+  @@if (is8285):
```

Comment:
Instead of duplicating this code except for the strings, I think it would be better to just set a variable with the expected file extension and the message to display, then the code stays the same on both platforms.
```diff
index 5bc66502..0f6e5d0a 100644
--- a/src/html/scan.js
+++ b/src/html/scan.js
@@ -513,29 +513,25 @@ function fileDragHover(e) {

 function fileSelectHandler(e) {
   fileDragHover(e);
+  // ESP32 expects .bin, ESP8285 RX expect .bin.gz
   const files = e.target.files || e.dataTransfer.files;
   const fileExt = files[0].name.split('.').pop();
-  @@if (is8285):
-  if (fileExt === 'gz') {
-    uploadFile(files[0]);
-  } else {
-    cuteAlert({
-      type: 'error',
-      title: 'Incorrect File Format',
-      message: 'You selected the file &quot;' + files[0].name.toString() + '&quot;.<br />The firmware file must be a .bin.gz file. <br />Do NOT decompress/unzip/extract the file!'
-    });
-  }
+@@if (is8285 and not isTX):
+  const expectedFileExt = 'gz';
+  const expectedFileExtDesc = '.bin.gz file. <br />Do NOT decompress/unzip/extract the file!';
 @@else:
-  if (fileExt === 'bin') {
+  const expectedFileExt = 'bin';
+  const expectedFileExtDesc = '.bin file.';
+@@endif
+  if (fileExt === expectedFileExt) {
     uploadFile(files[0]);
   } else {
     cuteAlert({
       type: 'error',
       title: 'Incorrect File Format',
-      message: 'You selected the file &quot;' + files[0].name.toString() + '&quot;.<br />The firmware file must be a .bin file.'
+      message: 'You selected the file &quot;' + files[0].name.toString() + '&quot;.<br />The firmware file must be a ' + expectedFileExtDesc
     });
   }
-  @@end
 }
```

Reply:
thanks Cap'n!